code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/ApiAuthenticator.java

```diff
@@ -150,7 +154,7 @@ protected Account getAccount(ApiRequest request) {
         for (IdentityTransformationHandler identityTransformationHandler : identityTransformationHandlers) {
             identities.addAll(identityTransformationHandler.getIdentities(account));
         }
-
+        identities.remove(null);
```

Comment:
This line looks a bit off; if we want to restrict the set from having not null elements, we can put a not null check at the place where element gets added to the set.